### DISCUSSION TITLE
User feedback: documentation could be better organized

### DISCUSSION THREAD

**Discussion by pedrolvg** [sqlpage/SQLPage#268] [General]
I am making a minimal admin panel with sqlpage and I am comming to the conclusion that documentation could benifit from another paradigm.

I love the fact that the webpage uses sqlpage itself, but I can't help to think that that might be holding back documentation quality.

I have found *most* of the things I am looking for, and once I find them, they are well explained. However it is not trivial to find them. For example, I am looking into how to add custom css or how to add static assets, and I can't seem to find it.

The components page is where most of the stuff I wanted to know was. But I have never thought writting components is actually a basic workflow, on which many things rely. 

In my opinion, this projects maturity and usefullness are already calling for the next step documentation wise.

A linear document with a structure of chapters and subchapters like postgresql documentation would provide a straightforward way to quickly find a bit of information.
Perhaps its time to swith to another documentation system. I would suggest something in the lines of [mdbook](https://rust-lang.github.io/mdBook/index.html) or similar

Edit: another suggestions that I am sure many people would find usefull and use often: a cookbook page, often times also called FAQs.
There is the examples folder, but one needs to dig into the files and hope to find a useful example.. The examples in the readme are a good start but they only have the code and an image. A list with a textual tittle and a paragraph or two, would be more usefull. Example of a list of things that I needed to chase that could be place in a cookbook:

 * serve static files
 * style a table with css
 * modify the main page design (shell component)
 * add scripts and stylesheets
 * simple CRUD app
 * write a simple component
 * deploy to aws lambda
 * handle form input
 * navigation from a list
 * add top navigation menu

**Comment by lovasoa** [comment]
Hello!

 This topic has come up before, and you are not the only one disappointed by the quality, quantity, and organization of the documentation on the official website. It's hard to make something that is accessible to both non-developers who are making their first app and to advanced users that want custom html, css, and js; but I'm sure we can do much better. 

https://github.com/lovasoa/SQLpage/issues/266
https://github.com/lovasoa/SQLpage/discussions/262

I'm in favor of doing as much as possible using SQLPage itself, and extending SQLPage itself when we need things we cannot do yet for the documentation. SQLPage does support the inclusion of markdown files, so this shouldn't prevent writing a lot of the documentation in individual .md files now. 

Would you be interested to write small markdown guides about some of the topics you mentioned above?

**Reply by pedrolvg** [reply]
I suppose the contents could even be saved in a database table and served with sqlpage. But truth be told, the format and the details are not the most important now as content can easily be converted or even copy pasted.

I can start putting together a mardownfile It's a good time now while I am still learning and have a sense of what is difficult to figure out.

**Reply by lovasoa** [reply]
Great, thank you ! Indeed, something focusing on what you found the most difficult and counter-intuitive would be the most useful.

I agree the technical details matter less, you can even post your explanations here, I'll upload and format them myself.
